client/ayon_maya/plugins/publish/extract_layout.py

```diff
@@ -176,6 +180,32 @@ def process(self, instance):
         self.log.debug("Extracted instance '%s' to: %s",
                        instance.name, json_representation)
 
+    def get_containers_sub_assembly(self, containers, container_root):
+        """Allow to collect the asset containers through sub-assembly workflow if
+        there is a layout container
+
+        Args:
+            containers (list): Ayon asset containers
+            container_root (str): the root of the asset container
+
+        Returns:
+            dict: container dict with the container and its related asset name
+        """
+        all_containers_set = {}
+        for existing_container in containers:
+            if cmds.getAttr(f"{existing_container}.loader").lower() == "layoutloader":
```

Comment:
Could a layout loader - technically contain another layout loader, if so - this may need to just be recursive?

Reply:
haven't thought of that before. Implement in [461e648](https://github.com/ynput/ayon-maya/pull/189/commits/461e64877655fe69afd7c2e2276bce8b975fd8e8)